opm/autodiff/StandardWells_impl.hpp

```diff
@@ -1248,4 +1248,300 @@ namespace Opm
     }
 
 
+
+
+
+
+   template<class WellState>
+   void
+   StandardWells::
+   updateListEconLimited(ScheduleConstPtr schedule,
+                         const int current_step,
+                         const Wells* wells_struct,
+                         const WellState& well_state,
+                         DynamicListEconLimited& list_econ_limited) const
+   {
+       const int nw = wells_struct->number_of_wells;
+
+       for (int w = 0; w < nw; ++w) {
+           // flag to check if the mim oil/gas rate limit is violated
+           bool rate_limit_violated = false;
+           const std::string& well_name = wells_struct->name[w];
+           const Well* well_ecl = schedule->getWell(well_name);
+           const WellEconProductionLimits& econ_production_limits = well_ecl->getEconProductionLimits(current_step);
+
+           // economic limits only apply for production wells.
+           if (wells_struct->type[w] != PRODUCER) {
+               continue;
+           }
+
+           // if no limit is effective here, then continue to the next well
+           if ( !econ_production_limits.onAnyEffectiveLimit() ) {
+               continue;
+           }
+           // for the moment, we only handle rate limits, not handling potential limits
+           // the potential limits should not be difficult to add
+           const WellEcon::QuantityLimitEnum& quantity_limit = econ_production_limits.quantityLimit();
+           if (quantity_limit == WellEcon::POTN) {
+               const std::string msg = std::string("POTN limit for well ") + well_name + std::string(" is not supported for the moment. \n")
+                                     + std::string("All the limits will be evaluated based on RATE. ");
+               OpmLog::warning("NOT_SUPPORTING_POTN", msg);
+           }
+
+           const WellMapType& well_map = well_state.wellMap();
+           typename WellMapType::const_iterator i_well = well_map.find(well_name);
```

Comment:
A suggestion: instead of dealing with this iterator, it would be better to deal with the value type (WellState::mapentry_t), especially as function parameters: you always take i_well->second whenever you use this object, so that value is all you need.